---
title: "1. Project Updates "
author: "Core Team"
output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 2
vignette: >
  % \VignetteIndexEntry{1. Project Updates}
  % \VignetteEngine{knitr::rmarkdown}
---
```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=100, max.print=1000)
options(useFancyQuotes=FALSE)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")), 
    error=FALSE)
```


Date: 20 July, 2015<br />

# _Bioconductor_ Project Updates

## Valerie Obenchain, including Herve Pages

### Valerie
- `BiocParallel`
  - unify interface for error handling, re-compting failed results
  - logging, worker timeout, write out results (`SnowParam`, `MulticoreParam`)
- `GenomicFiles`
- coordinate mapping:
  - `mapToTranscripts()`, `mapFromTranscripts()` (`GenomicFeatures` package)
  - `mapToAlignments()`, `mapFromAlignments()` (`GenomicAlignments` package)
- Newsletters

### Herve

- NCList: Nested Containment List algorithm for counting overlaps
  - 'algorithm' arg added to overlaps operations
- `BSGenome` package
  - `XtraSNPlocs` class and packages
    hold locations of _in-del_, _heterozygous_, _microsatellite_, _named-locus_, _no-variation_, _mixed_, etc.
  - `snpsBySeqname()`, `snpsByOverlaps()`, `snpsById()` methods on `SNPlocs`, `XtraSNPlocs` objects
- `makeGRangesFromDataFrame()`, `makeTxDbFrom*()` functions
- refactor `SummarizedExperiment` class


## Marc Carlson

## Dan Tenenbaum

Since BioC2014:

* Mavericks Build Machines (Snow Leopard support to be dropped soon)
* Support Site went live
* [Search course materials by topic](http://bioconductor.org/help/course-materials/); Course AMIs available
* Began Docker support (attend my workshop for more info)
* Phasing out Git-Svn Bridge in favor of Git Mirrors
* Use/usability shields on package landing pages

## Sonali Arora

- __AnnotationHub__
   - Rewrite recipes after Major AnnotationHub refactor
   - New resources added to the hub 
      - GTF files from Ensembl release 76 to 80 (available as GRanges object)
      - [Epigenome RoadMap Files](http://www.roadmapepigenomics.org/)
      - [UCSC Chain files](http://genome.ucsc.edu/goldenpath/help/chain.html)
      - [UCSC Two Bit Files](http://genome.ucsc.edu/FAQ/FAQformat.html#format7)
      - [BioPAX files](http://pid.nci.nih.gov/download.shtml) (Level1 and Level2) from the NCI Pathway Interaction Database 
      - [Pazar](http://www.pazar.info/tftargets/) - Database of Transcription Factor and  Regulatory Sequence Analysis
      - Background file for [ChEA](http://amp.pharm.mssm.edu/lib/chea.jsp) required for the command line version of ChEA (available as data.frame object)
   - propogate metadata along with the object, once extracted from the Hub.
   - show() method for AnnotationHub (with Martin)  
   - Tutorial on [Epigenome RoadMap Files](http://www.bioconductor.org/packages/devel/bioc/vignettes/AnnotationHub/inst/doc/AnnotationHub-HOWTO.html)
   - Video on [AnnotationHub](https://www.youtube.com/watch?v=pFvUOPfR8eA&feature=youtu.be)
- Taken over as __Package Moderator__ from Marc Carlson since Jan 2015
    - approcimately 120 packages added
    - added 'pre-accepted' Stage to review process 
    - Google Hangout for New Package Authors with Marc to share some key development tips and solicit questions from a wider audience.
- __biocViews__
    - New function - biocViews::recommendPackages(), searches release and devel
    - Revise and Reorganize [ExperimentData](http://www.bioconductor.org/packages/devel/BiocViews.html#___ExperimentData) branch of biocViews
    - added new biocViews
        - ExperimentData -> [ReproducibleResearch](http://www.bioconductor.org/packages/devel/BiocViews.html#___ReproducibleResearch) to highlight packages containing code from Publications/Books
        - ExperimentData -> RepositoryData -> [Project1000genomes](http://www.bioconductor.org/packages/devel/BiocViews.html#___Project1000genomes)
        - Software -> WorkflowStep -> Pathways -> [KEGG](http://www.bioconductor.org/packages/devel/BiocViews.html#___KEGG), [Reactome](http://www.bioconductor.org/packages/devel/BiocViews.html#___Reactome), [BioCarta](http://www.bioconductor.org/packages/devel/BiocViews.html#___BioCarta), [NCINatureCurated](http://www.bioconductor.org/packages/devel/BiocViews.html#___NCINatureCurated) 
        - ExperimentData -> TechnologyData -> [HighThroughputImagingData](http://www.bioconductor.org/packages/devel/BiocViews.html#___HighThroughputImagingData)
        - ExperimentData -> SpecimenSource -> [Somatic](http://www.bioconductor.org/packages/devel/BiocViews.html#___Somatic), [Germline](http://www.bioconductor.org/packages/devel/BiocViews.html#___Germline), [Proteome](http://www.bioconductor.org/packages/devel/BiocViews.html#___Proteome)
        - AnnotationData -> [CodeLinkChip](http://www.bioconductor.org/packages/devel/BiocViews.html#___CodelinkChip) 
        - Software -> Technology -> Sequencing -> [HiC](http://www.bioconductor.org/packages/devel/BiocViews.html#___HiC)
        - Software -> WorkflowStep -> Alignment -> [MultipleSequenceAlignment](http://www.bioconductor.org/packages/devel/BiocViews.html#___MultipleSequenceAlignment)
        - Software -> ResearchField -> [Transcriptomics](http://www.bioconductor.org/packages/devel/BiocViews.html#___Transcriptomics) 
- __Revise__ [Sequencing Workflow](http://www.bioconductor.org/help/workflows/high-throughput-sequencing/)     
- __Assist/Teach Workshops__ with Martin
   - [SeattleOct2014](http://bioconductor.org/help/course-materials/2014/SeattleOct2014/)
   - [SeattleApr2015](http://bioconductor.org/help/course-materials/2015/SeattleApr2015/)
   - [CSAMA2015](http://www-huber.embl.de/csama/)
   - [useR2015](http://user2015.math.aau.dk/#BN)
- __Website Design__, Implementation by Dan
   - Course Material Page


## Paul Shannon

## Jim Hester

- Git/GitHub Mirrors for SVN repositories (https://github.com/Bioconductor-mirror/)
- Test Coverage Shields (package landing pages and https://codecov.io/github/Bioconductor-mirror)
- `SummarizedExperiment`/`ExpressionSet` coercion methods and `rowRanges()` accessor.

## Martin Morgan, for Nathaniel Hayden

- Rhtslib

  - GNUlib is neat
  - bamsignals -- 'straight-forward' use; good example
  - deepSNV -- takes advantage of CRAM support

- Rsamtools with Rhtslib

  - To be released immediately after BioC2015

- h5robj -- S4 in h5
